diff --git a/submodules/AccountContext/Sources/Premium.swift b/submodules/AccountContext/Sources/Premium.swift
index 3bb1f6c..b7d664d 100644
--- a/submodules/AccountContext/Sources/Premium.swift
+++ b/submodules/AccountContext/Sources/Premium.swift
@@ -150,7 +150,7 @@ public enum StarsPurchasePurpose: Equatable {
 public struct PremiumConfiguration {
     public static var defaultValue: PremiumConfiguration {
         return PremiumConfiguration(
-            isPremiumDisabled: false,
+            isPremiumDisabled: true,
             areStarsDisabled: true,
             subscriptionManagementUrl: "",
             showPremiumGiftInAttachMenu: false,
@@ -273,14 +273,14 @@ public struct PremiumConfiguration {
                 return (value as? Double).flatMap(Int32.init)
             }
             return PremiumConfiguration(
-                isPremiumDisabled: data["premium_purchase_blocked"] as? Bool ?? defaultValue.isPremiumDisabled,
-                areStarsDisabled: data["stars_purchase_blocked"] as? Bool ?? defaultValue.areStarsDisabled,
+                isPremiumDisabled: true,
+                areStarsDisabled: true,
                 subscriptionManagementUrl: data["premium_manage_subscription_url"] as? String ?? "",
-                showPremiumGiftInAttachMenu: data["premium_gift_attach_menu_icon"] as? Bool ?? defaultValue.showPremiumGiftInAttachMenu,
-                showPremiumGiftInTextField: data["premium_gift_text_field_icon"] as? Bool ?? defaultValue.showPremiumGiftInTextField,
-                giveawayGiftsPurchaseAvailable: data["giveaway_gifts_purchase_available"] as? Bool ?? defaultValue.giveawayGiftsPurchaseAvailable,
-                starsGiftsPurchaseAvailable: data["stars_gifts_enabled"] as? Bool ?? defaultValue.starsGiftsPurchaseAvailable,
-                starGiftsPurchaseBlocked: data["stargifts_blocked"] as? Bool ?? defaultValue.starGiftsPurchaseBlocked,
+                showPremiumGiftInAttachMenu: false,
+                showPremiumGiftInTextField: false,
+                giveawayGiftsPurchaseAvailable: false,
+                starsGiftsPurchaseAvailable: false,
+                starGiftsPurchaseBlocked: true,
                 boostsPerGiftCount: get(data["boosts_per_sent_gift"]) ?? defaultValue.boostsPerGiftCount,
                 audioTransciptionTrialMaxDuration: get(data["transcribe_audio_trial_duration_max"]) ?? defaultValue.audioTransciptionTrialMaxDuration,
                 audioTransciptionTrialCount: get(data["transcribe_audio_trial_weekly_number"]) ?? defaultValue.audioTransciptionTrialCount,
diff --git a/submodules/TelegramCore/Sources/TelegramEngine/Messages/AdMessages.swift b/submodules/TelegramCore/Sources/TelegramEngine/Messages/AdMessages.swift
index d8d8ffd..52f4606 100644
--- a/submodules/TelegramCore/Sources/TelegramEngine/Messages/AdMessages.swift
+++ b/submodules/TelegramCore/Sources/TelegramEngine/Messages/AdMessages.swift
@@ -635,18 +635,7 @@ public class AdMessagesHistoryContext {
     public let messageId: EngineMessage.Id?
     
     public var state: Signal<(interPostInterval: Int32?, messages: [Message], startDelay: Int32?, betweenDelay: Int32?), NoError> {
-        return Signal { subscriber in
-            let disposable = MetaDisposable()
-            
-            self.impl.with { impl in
-                let stateDisposable = impl.state.get().start(next: { state in
-                    subscriber.putNext((state.interPostInterval, state.messages, state.startDelay, state.betweenDelay))
-                })
-                disposable.set(stateDisposable)
-            }
-            
-            return disposable
-        }
+        return .single((interPostInterval: nil, messages: [], startDelay: nil, betweenDelay: nil))
     }
     
     public init(account: Account, peerId: EnginePeer.Id, messageId: EngineMessage.Id? = nil, activateManually: Bool = false) {
@@ -655,7 +644,7 @@ public class AdMessagesHistoryContext {
         
         let queue = self.queue
         self.impl = QueueLocalObject(queue: queue, generate: {
-            return AdMessagesHistoryContextImpl(queue: queue, account: account, peerId: peerId, messageId: messageId, activateManually: activateManually)
+            return AdMessagesHistoryContextImpl(queue: queue, account: account, peerId: peerId, messageId: messageId, activateManually: true)
         })
     }
 
diff --git a/submodules/TelegramCore/Sources/TelegramEngine/Peers/AdPeers.swift b/submodules/TelegramCore/Sources/TelegramEngine/Peers/AdPeers.swift
index ea6bb2d..f7f76c4 100644
--- a/submodules/TelegramCore/Sources/TelegramEngine/Peers/AdPeers.swift
+++ b/submodules/TelegramCore/Sources/TelegramEngine/Peers/AdPeers.swift
@@ -39,60 +39,5 @@ public class AdPeer: Equatable {
 }
 
 func _internal_searchAdPeers(account: Account, query: String) -> Signal<[AdPeer], NoError> {
-    return account.network.request(Api.functions.contacts.getSponsoredPeers(q: query))
-    |> map(Optional.init)
-    |> `catch` { _ in
-        return .single(nil)
-    }
-    |> mapToSignal { result in
-        guard let result else {
-            return .single([])
-        }
-        return account.postbox.transaction { transaction -> [AdPeer] in
-            switch result {
-            case let .sponsoredPeers(sponsoredPeersData):
-                let (peers, chats, users) = (sponsoredPeersData.peers, sponsoredPeersData.chats, sponsoredPeersData.users)
-                let parsedPeers = AccumulatedPeers(transaction: transaction, chats: chats, users: users)
-                updatePeers(transaction: transaction, accountPeerId: account.peerId, peers: parsedPeers)
-                
-                var subscribers: [PeerId: Int32] = [:]
-                for chat in chats {
-                    if let groupOrChannel = parseTelegramGroupOrChannel(chat: chat) {
-                        switch chat {
-                        case let .channel(channelData):
-                            let participantsCount = channelData.participantsCount
-                            if let participantsCount = participantsCount {
-                                subscribers[groupOrChannel.id] = participantsCount
-                            }
-                        default:
-                            break
-                        }
-                    }
-                }
-                
-                var result: [AdPeer] = []
-                for peer in peers {
-                    switch peer {
-                    case let .sponsoredPeer(sponsoredPeerData):
-                        let (randomId, apiPeer, sponsorInfo, additionalInfo) = (sponsoredPeerData.randomId, sponsoredPeerData.peer, sponsoredPeerData.sponsorInfo, sponsoredPeerData.additionalInfo)
-                        guard let peer = parsedPeers.get(apiPeer.peerId) else {
-                            continue
-                        }
-                        result.append(
-                            AdPeer(
-                                opaqueId: randomId.makeData(),
-                                peer: EnginePeer(peer),
-                                subscribers: subscribers[peer.id],
-                                sponsorInfo: sponsorInfo,
-                                additionalInfo: additionalInfo
-                            )
-                        )
-                    }
-                }
-                return result
-            default:
-                return []
-            }
-        }
-    }
+    return .single([])
 }
diff --git a/submodules/TelegramCore/Sources/Utils/PeerUtils.swift b/submodules/TelegramCore/Sources/Utils/PeerUtils.swift
index 7db7444..3b5d53b 100644
--- a/submodules/TelegramCore/Sources/Utils/PeerUtils.swift
+++ b/submodules/TelegramCore/Sources/Utils/PeerUtils.swift
@@ -207,12 +207,7 @@ extension Peer {
     }
     
     var isPremium: Bool {
-        switch self {
-        case let user as TelegramUser:
-            return user.flags.contains(.isPremium)
-        default:
-            return false
-        }
+        return false
     }
     
     var isSubscription: Bool {
